src/stellar/StellarTools.ts

```diff
@@ -81,11 +87,33 @@ export class StellarTools implements EdgeCurrencyTools {
     return { stellarKey: keypair.secret() }
   }
 
-  async importPrivateKey(privateKey: string): Promise<{ stellarKey: string }> {
-    privateKey.replace(/ /g, '')
-    stellarApi.Keypair.fromSecret(privateKey)
-    if (privateKey.length !== 56) throw new Error('Private key wrong length')
-    return await Promise.resolve({ stellarKey: privateKey })
+  async importPrivateKey(userInput: string): Promise<StellarPrivateKeys> {
+    if (/^(0x)?[0-9a-fA-F]{64}$/.test(userInput)) {
+      // It looks like a private key, so validate the hex:
+      userInput.replace(/ /g, '')
+      stellarApi.Keypair.fromSecret(userInput)
+      if (userInput.length !== 56) throw new Error('Private key wrong length')
+
+      return await Promise.resolve({
+        stellarKey: userInput,
+        stellarMnemonic: ''
+      })
+    } else {
+      // it looks like a mnemonic, so validate that way:
+      if (!validateMnemonic(userInput)) {
+        // "input" instead of "mnemonic" in case private key
+        // was just the wrong length
+        throw new Error('Invalid input')
+      }
+      const derivedSeed = ed25519.derivePath("m/44'/314159'/0'", userInput).key
```

Comment:
Where did you get this path? The XLM SLIP44 path is 148.

Reply:
https://github.com/limitpointinf0/pi-wallet-cli/blob/e9c016e0d686aa04fb9ddf2411d49b52d4b82511/src/createAsset.js#L44

Reply:
And the code doesn't work. `derivePath` doesn't accept a mnemonic directly it takes the hex encoded version. You need to use bip39's `mnemonicToSeed` like pi-wallet-cli and SolanaTools does

Reply:
I'm confused, pi-wallet-cli also uses the `m/44'/314159'/0'`` path when deriving
from mnemonic:
https://github.com/limitpointinf0/pi-wallet-cli/blob/e9c016e0d686aa04fb9ddf2411d49b52d4b82511/src/piLib.js#L22